keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -32,122 +33,210 @@ superseded-by:
+ name prefixing / suffixing, labeling, annotating, namespacing, imageTag setting.
+- Performing flexible targeted transformations intended to be applied to specific Resource Configs - e.g.
+ strategic patch, json patch
+- Composing and layering Resource Config and schema-aware transformations
+- Facilitating rolling updates to Resources such as ConfigMaps and Secrets
+- Facilitating creation of Resources that require the creation to be ordered - e.g. namespaces, crds
 
-commonAnnotations:
- oncallPager: 800-555-1212
+### Non-Goals
 
-configMapGenerator:
-- name: myJavaServerEnvVars
- literals:   
- - JAVA_HOME=/opt/java/jdk
- - JAVA_TOOL_OPTIONS=-agentlib:hprof
+- Exposing all imperative kubectl commands as declarative options
+- Exposing all kubectl generators and schema-aware transformations
+- Providing simpler alternatives to the APIs for declaring Resource Config (e.g. a simple way to create deployments)
+- Providing a templating mechanism (e.g. for generating Resource Config)
+- Publishing with kubectl any other Kustomize sub-commands besides `build`
 
-secretGenerator:
-- name: app-sec
- commands:
-   username: "echo admin"
-   password: "echo secret"
-```
-The build output of this sample kustomizaiton.yaml file is
-```
-apiVersion: v1
-data:
- JAVA_HOME: /opt/java/jdk
- JAVA_TOOL_OPTIONS: -agentlib:hprof
-kind: ConfigMap
-metadata:
- annotations:
-   oncallPager: 800-555-1212
- name: alices-myJavaServerEnvVars-7bc9c27cmf
----
-apiVersion: v1
-data:
- password: c2VjcmV0Cg==
- username: YWRtaW4K
-kind: Secret
-metadata:
- annotations:
-   oncallPager: 800-555-1212
- name: alices-app-sec-c7c5tbh526
-type: Opaque
-```
+### Why should this be part of kubectl?
+
+- It was purposefully built to address user friction in kubectl declarative workflows. Leaving these issues unaddressed
+ in the command itself reduces the quality of the product.
+- The techniques it uses to address the issues are based on the existing kubectl imperative commands.  It is
+ consistent with the rest of kubectl.
+- Bridging the imperative and declarative workflows helps bring a more integrated feel and consistent story to
+ kubectl's command set.
+- Kustomize is already part of the Kubernetes project and owned by SIG CLI (the same sig that owns kubectl).
+- SIG CLI have expertise in the Kustomize codebase and is committed to maintaining this solution going forward.
+- Providing it as part of kubectl will ensure that it is available to users of kubectl apply.
 
 ## Proposal
 
-### UX
+Do one or more of these.
+
+- [ ] A: `kubectl kustomize -f dir/ | kubectl apply -f -`
+- [ ] B: `kubectl apply -f dir/`
+- [ ] C: `kubectl apply -f dir/kustomization.yaml` (similar to B, but slightly different UX)
+
+### Option A: Kustomize Command
+
+Publish the `kustomize build` command as `kubectl kustomize`.  Update documentation demonstrate using Kustomize as
+`kubectl kustomize -f dir/ | kubectl kustomize -f -`.  Consider Option B as a follow up if there is consensus on
```

Comment:
The second "kustomize" here should be "apply:"
`kubectl kustomize -f dir/ | kubectl apply -f -`.

It is unfortunate that this requires two separate invocations of _kubectl_.